### ISSUE TITLE
Higher train loss and worse evaluation metrics when using `torch.compile()`

### ISSUE THREAD

**Issue by snarayan21** [pytorch/pytorch#113180]
### 🐛 Describe the bug

We are facing issues with loss curves and reproducibility when using `torch.compile()` with our models. Attached below is a graph of train loss with runs with `torch.compile()` (higher loss) and runs without (lower loss). This model is an MPT-style transformer, but we've also seen the issue occur with evaluation for an autoencoder setup (also shown below). Would love to address this issue as soon as possible!

Higher train loss:
<img width="1623" alt="Screenshot 2023-11-02 at 8 28 39 AM" src="https://github.com/pytorch/pytorch/assets/28932043/bdb58b54-8027-4ff9-a514-eb98a69cce13">

Worse eval scores (orange and turquoise are with `torch.compile()`:
![Screenshot 2023-11-02 at 10 15 57 PM (1)](https://github.com/pytorch/pytorch/assets/28932043/cbb55c2f-8c95-4eac-b52c-706d733ebba2)

### Error logs

We're not seeing error logs, but the loss and evaluation metrics for the models are very different.

### Minified repro

```
import torch
from torch import tensor, device
import torch.fx as fx
from torch._dynamo.testing import rand_strided
from math import inf
import torch._inductor.inductor_prims

import torch._dynamo.config
import torch._inductor.config
import torch._functorch.config
torch._dynamo.config.automatic_dynamic_shapes = False
torch._dynamo.config.suppress_errors = True
torch._inductor.config.fallback_random = True
torch._inductor.config.generate_intermediate_hooks = True



isolate_fails_code_str = None



# torch version: 2.1.0+cu121
# torch cuda version: 12.1
# torch git version: 7bcf7da3a268b435777fe87c7794c382f444e86d


# CUDA Info: 
# nvcc: NVIDIA (R) Cuda compiler driver 
# Copyright (c) 2005-2023 NVIDIA Corporation 
# Built on Mon_Apr__3_17:16:06_PDT_2023 
# Cuda compilation tools, release 12.1, V12.1.105 
# Build cuda_12.1.r12.1/compiler.32688072_0 

# GPU Hardware Info: 
# NVIDIA A100-SXM4-40GB : 4 


from torch.nn import *
class Repro(torch.nn.Module):
    def __init__(self):
        super().__init__()

    
    
    def forward(self, permute):
        return [permute]
        
def load_args(reader):
    buf0 = reader.storage('c836a62b3bfef0f83a406d6327f6af0cf3833814', 50331648, device=device(type='cuda', index=1), dtype_hint=torch.bfloat16)
    reader.tensor(buf0, (2, 2048, 48, 128), dtype=torch.bfloat16, is_leaf=True)  # permute
load_args._version = 0
mod = Repro()
if __name__ == '__main__':
    from torch._dynamo.repro.after_aot import run_repro
    run_repro(mod, load_args, accuracy=True, command='run', save_dir='/mnt/workdisk/saaketh/torch_compile_debug/run_2023_11_06_01_50_22_131795-pid_97282/minifier/checkpoints', tracing_mode='real', check_str=None)
```

### Versions

PyTorch version: 2.2.0a0+git21b6030
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: version 3.26.3
Libc version: glibc-2.31

Python version: 3.10.13 (main, Aug 25 2023, 13:20:03) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.4.0-137-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 12.1.105
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA A100-SXM4-40GB
GPU 1: NVIDIA A100-SXM4-40GB

Nvidia driver version: 515.48.07
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   48 bits physical, 48 bits virtual
CPU(s):                          64
On-line CPU(s) list:             0-63
Thread(s) per core:              1
Core(s) per socket:              32
Socket(s):                       2
NUMA node(s):                    8
Vendor ID:                       AuthenticAMD
CPU family:                      25
Model:                           1
Model name:                      AMD EPYC 7513 32-Core Processor
Stepping:                        1
Frequency boost:                 enabled
CPU MHz:                         1435.899
CPU max MHz:                     2600.0000
CPU min MHz:                     1500.0000
BogoMIPS:                        5199.66
Virtualization:                  AMD-V
L1d cache:                       2 MiB
L1i cache:                       2 MiB
L2 cache:                        32 MiB
L3 cache:                        256 MiB
NUMA node0 CPU(s):               0-7
NUMA node1 CPU(s):               8-15
NUMA node2 CPU(s):               16-23
NUMA node3 CPU(s):               24-31
NUMA node4 CPU(s):               32-39
NUMA node5 CPU(s):               40-47
NUMA node6 CPU(s):               48-55
NUMA node7 CPU(s):               56-63
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 invpcid_single hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold v_vmsave_vmload vgif umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca

Versions of relevant libraries:
[pip3] numpy==1.26.0
[pip3] onnx==1.14.0
[pip3] onnxruntime==1.15.1
[pip3] optree==0.9.2
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-triton==2.1.0+6e4932cda8
[pip3] torch==2.2.0a0+git21b6030
[pip3] torch-optimizer==0.3.0
[pip3] torchmetrics==1.0.3
[pip3] torchvision==0.16.0+cu121
[pip3] triton-nightly==2.1.0.dev20230726014945
[pip3] triton-pre-mlir==2.0.0
[conda] Could not collect

**Comment by snarayan21** [comment]
Hey, is there anything else I can provide for this issue? This is a major issue we're seeing for many of our models at this point.

**Comment by ezyang** [comment]
The minifier script is not helpful. Are you able to run some ablation experiments; e.g., can you try with `backend="aot_eager"` and see if converges that way?

**Comment by voznesenskym** [comment]
@snarayan21 please try as directed

**Comment by snarayan21** [comment]
Hey @ezyang @voznesenskym apologies for the delay in getting around to this. I just ran with `backend="aot_eager"` on a smaller model and this does converge (no compile: orange, overlaps with aot_eager in blue, and without aot_eager shows no change in train loss):
<img width="1599" alt="Screenshot 2023-11-20 at 12 36 32 PM" src="https://github.com/pytorch/pytorch/assets/28932043/4f516061-71c5-42e9-a848-5eabb274f604">

According to [this page](https://pytorch.org/docs/stable/torch.compiler_troubleshooting.html) the issue is with TorchInductor, but how would I go about root-causing this?

Thank you for your help!

**Comment by ezyang** [comment]
You could try running the accuracy minifier; chances are it's not going to work, but sometimes you get lucky. https://pytorch.org/docs/stable/torch.compiler_troubleshooting.html

A full set of debug logs, ala TORCH_LOGS=+dynamo,+aot,+inductor may help. If you have instructions to reproduce the training that might help too. It converging on aot_eager is a clear indication that it's an Inductor problem. If you can try `aot_eager_decomp_partition` that also will give more signal if it's a decomp problem.

**Comment by snarayan21** [comment]
I just tested this with torch 2.2.0 and the issue persists (see below). I previously did use the accuracy minifier -- would you recommend using that with `backend="aot_eager"`? Or is there another way to diagnose what's being compiled wrong?
<img width="1621" alt="Screenshot 2023-12-15 at 5 09 04 PM" src="https://github.com/pytorch/pytorch/assets/28932043/98a52743-9135-4096-85c0-c49ff443638b">

**Comment by Skylion007** [comment]
Okay, I confirmed the error does still happen with `aot_eager_decomp_partition` suggesting that it may be a decomp problem. How do we debug further?

**Comment by bdhirsh** [comment]
@Skylion007 were you able to repro locally? The repro given above looks like a failure in the minifier.

A repro would help a bunch with narrowing down further. A few things I would try next if I could repro are:

(1) also run with `backend="aot_eager"`.:

(1a) if it passes, then... there are still a couple options, but one likely culprit is one of the inductor decomps, which are run in `aot_eager_decomp_partition` but not `aot_eager`. You could bisect them by removing decomps from [here](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/decomposition.py#L36).

(1b) If it fails, but `compile(backend="eager")` passes, then there are also a few options: AOTAutograd bug, functionalization bug, custom ops issues, and a few others. In this case, one useful thing to check would be if there are any (non-ATen) custom operators in your model.

**Comment by Skylion007** [comment]
@bdhirsh I was able to repro locally, and I some rough repo have instructions now and it should repro in as little as half an hour of training:
1. Follow the instructions to train the MosaicML BERT model here on C4: https://github.com/mosaicml/examples/tree/main/examples/benchmarks/bert
2. Add the kwarg to this trainer: https://github.com/mosaicml/examples/blob/7003793b15ad0ee28bc09d0fceb91eb2d0104961/examples/benchmarks/bert/main.py#L248  `compile_config = {"backend": "aot_eager_decomp_partition"}`. 
3. Compare results with BERT with the compile_config set to None and the compile_config. You will notice that they behave very different. If you have an 8A100 machine, you should notice divergence should be obvious in as little as a 1000 batches. 

Let me know if you need more details to repro.

**Comment by bdhirsh** [comment]
@Skylion007  I haven't been able to fully repro. Pasting what I've done so far + current issues below:

** Stuff I did so far **

Mostly just followed the readme steps + hit a few snags (jotting them down here)

* comment out an sklearn import due to https://github.com/pybind/pybind11/discussions/3453 (I don't seem to have `GLIBCXX_3.4.29` on my machine)

* upgrade my datasets library due to https://stackoverflow.com/questions/77433096/notimplementederror-loading-a-dataset-cached-in-a-localfilesystem-is-not-suppor

* I had a newer version of triton, so had to manually fix some bits of the triton flash attention impl used in bert: https://github.com/openai/triton/issues/1098

* Had to follow readme to change the yaml from "split: train" to "split: train_small"

* Had to install apex

* This was enough to get `composer main.py yamls/main/mosaic-bert-base-uncased.yaml` running properly with eager mode. When I re-ran it a day later though, I got some weird shm errors - I worked around it by manually changing / copying over the data dir, `data_local: ./my-copy-c4`, to `data_local: ./my-copy2-c4`.


**Current issue**

`composer main.py yamls/main/mosaic-bert-base-uncased.yaml` is no longer running properly for me - I now get this issue:

```
Initializing model...
n_params=1.3740e+08
Building train loader...
Traceback (most recent call last):
  File "/data/users/hirsheybar/b/pytorch/examples/examples/benchmarks/bert/main.py", line 272, in <module>
    main(cfg)
  File "/data/users/hirsheybar/b/pytorch/examples/examples/benchmarks/bert/main.py", line 177, in main
    train_loader = build_dataloader(
  File "/data/users/hirsheybar/b/pytorch/examples/examples/benchmarks/bert/main.py", line 134, in build_dataloader
    return text_data_module.build_text_dataloader(cfg, tokenizer,
  File "/data/users/hirsheybar/b/pytorch/examples/examples/benchmarks/bert/src/text_data.py", line 274, in build_text_dataloader
    dataset = StreamingTextDataset(
  File "/data/users/hirsheybar/b/pytorch/examples/examples/benchmarks/bert/src/text_data.py", line 134, in __init__
    super().__init__(
  File "/home/hirsheybar/local/b/pytorch-env/lib/python3.10/site-packages/streaming/base/dataset.py", line 325, in __init__
    self._shm_prefix, self._locals_shm = get_shm_prefix(my_locals, world)
  File "/home/hirsheybar/local/b/pytorch-env/lib/python3.10/site-packages/streaming/base/shared.py", line 340, in get_shm_prefix
    raise ValueError(f'Reused local directory: {sorted(my_locals_set)} vs ' +
ValueError: Reused local directory: ['/data/users/hirsheybar/b/pytorch/examples/examples/benchmarks/bert/my-copy-c4/train_small'] vs ['/data/users/hirsheybar/b/pytorch/examples/examples/benchmarks/bert/my-copy-c4/train_small']. Provide a different one.
```

I'm not really sure how to interpret that error message. At one point I added a `breakpoint()` inside dynamo, which was probably a mistake since the repro is running a distributed harness (no idea if that's related to the issue that I'm now seeing though).

To be clear, I wasn't seeing that error message a few days ago but I am now. I tried this:
```
rm -rf ./my-copy-c4
python src/convert_dataset.py --dataset c4 --data_subset en --out_root ./my-copy-c4 --splits train_small val
```

But I'm getting the same error.

**Comment by Skylion007** [comment]
Okay yeah so you need to delete your `local` directory if you change the dataset at all or if you previous convert_dataset failed for any reason. so you probably just need to delete `/data/users/hirsheybar/b/pytorch/examples/examples/benchmarks/bert/my-copy-c4/train_small` and regenerate the dataset.  Essentially, it will not overwrite a local directory has a cached directory as this is usually an error, so you will need to just regenerate a new one.

I started new PR here that removes the triton dependency feel free to give it a whirl: https://github.com/mosaicml/examples/pull/440 and removes other problematic dependencies. You can also train wihtout the FlashAttention dependencies (on a way, way smaller batch size) and I suspect you will run into the same issue). You also do not need to install apex anymore, if you are PyTorch 2.0> you can switch the algorithm in the yaml to LowPrecisionLayerNorm instead. I will update the YAML in the PR to use that.

**Comment by Skylion007** [comment]
I just realized Stable Diffusion and BERT are both skipped in the latest benchmark tests  so it's possible the issue could be more widespread: https://github.com/pytorch/pytorch/blob/139c4ab69da404d4e0c0d099a728c74ce06e341b/benchmarks/dynamo/torchbench.py#L237

**Comment by soumith** [comment]
@bdhirsh's fix here might have also fixed this issue: https://github.com/pytorch/pytorch/issues/116935#issuecomment-1935115652

fingers crossed.

**Comment by chauhang** [comment]
@Skylion007 @snarayan21 Can you please test with the latest nightlies and see if the issue has been resolved?

**Comment by masnesral** [comment]
I'm helping scrub old issues this week. @Skylion007 @snarayan21 from the comment above, are one of you able to check if the issue has been resolved?

**Comment by Skylion007** [comment]
The BERT problem was solved more or less, SD2 hits a different issue with a bug at the intersection of mixed precision, torch compile, FSDP-1, and evalling after training for a bit and saving, but it does seem like the BERT problem is more or less fixed.

**Comment by Skylion007** [comment]
@snarayan21 Will try reproing the original issue later this week.

**Comment by alcinos** [comment]
I am not sure if OP's problem has been resolved or not, but in case it helps anyone finding this thread:
I have been facing similar issues on a different model and bleeding edge pytorch (2.5.1). One thing I noticed is that compiling the *module* is running into performance degradation as described in this thread, but compiling the *forward* does not (while still reaping most if not all of the speed-up).
I haven't been able to pinpoint the exact cause of this finding, but thought it may help someone out.

TLDR:
Don't do
```
model = torch.compile(model)
```

Rather do
```
class MyModel(nn.Module):
   def __init__(self, ....):
        ...... init code as usual .....
       self.forward = torch.compile(self.forward)
```

**Comment by bdhirsh** [comment]
@alcinos if by any chance you're able to share some kind of reproducer, I'd be happy to investigate more. If compiling `self.forward()` works for you then great but it's definitely a band-aid fix: in particular, you are going to miss out on any module hooks on `MyModel` getting compiled too.

**Comment by ae99** [comment]
Hey. I'm also running into a problem like this. Currently training a vanilla Llama architecture model and getting this same strange behavior with instability and nearly no loss change when compiled. Only appears for certain model sizes.

For me it appears the cause was a strange interaction in activation checkpointing.
```
x = checkpoint(block, x, input_pos, use_reentrant=False, context_fn=partial(create_selective_checkpoint_contexts, [torch.ops.aten._scaled_dot_product_flash_attention.default, torch.ops.aten._flash_attention_forward.default]), preserve_rng_state=False)
```
Removing the SDPA ops from the context resolved the divergence.

I actually ran into a very similar symptom previously and after lots of debugging found the issue being a small region with torch.no_grad misbehaving, https://github.com/pytorch/pytorch/issues/143632.

Would love to know if there's some way to catch these earlier with some sort of sanity check that compares eager activations/gradients for every layer. Certainly makes me paranoid that I'm only noticing when there's glaring problems but potentially missing this sort of occurrence on a smaller scale frequently.

**Comment by bdhirsh** [comment]
Thanks for digging @ae99. Any chance you have some repro code and/or a github repo with the checkpointing + SDPA issue that I can stare at?

**Comment by masnesral** [comment]
[scrubbing old issues] @ae99:

> Any chance you have some repro code and/or a github repo with the checkpointing + SDPA issue that I can stare at?

**Comment by dsrivastavv** [comment]
@bdhirsh @masnesral I have a repro with LLamaGen codebase (https://github.com/FoundationVision/LlamaGen). I am observing that if I compile with `inductor` backend, loss is not reducing fast compared to no-compile or `not_eager`. Code works fine with pytorch 2.3 but fails in 2.7.1 and 2.8 (haven't tested the intermediate versions). Please find details below:


no-compile
```
The following values were not passed to `accelerate launch` and had defaults used instead:
	`--num_processes` was set to a value of `2`
	`--num_machines` was set to a value of `1`
	`--mixed_precision` was set to a value of `'no'`
	`--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
None
None
/home/dsriv/miniconda3/envs/llamagen_dummy/lib/python3.12/site-packages/torch/distributed/distributed_c10d.py:4807: UserWarning: No device id is provided via `init_process_group` or `barrier `. Using the current device set by the user. 
  warnings.warn(  # warn only once
[rank1]:[W814 13:34:05.083016285 ProcessGroupNCCL.cpp:5023] [PG ID 0 PG GUID 0 Rank 1]  using GPU 1 as device used by this process is currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect. You can specify device_id in init_process_group() to force use of a particular device.
wandb: Tracking run with wandb version 0.21.1
wandb: W&B syncing is set to `offline` in this directory. Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
wandb: Run data is saved locally in outputs/llamagen_b/20250814_13/wandb/offline-run-20250814_133405-1jqb8z7w
wandb:                                                                                
wandb: You can sync this run to the cloud by running:
wandb: wandb sync outputs/llamagen_b/20250814_13/wandb/offline-run-20250814_133405-1jqb8z7w
wandb: Find logs at: outputs/llamagen_b/20250814_13/wandb/offline-run-20250814_133405-1jqb8z7w/logs
wandb: Tracking run with wandb version 0.21.1
wandb: W&B syncing is set to `offline` in this directory. Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
wandb: Run data is saved locally in outputs/llamagen_b/20250814_13/wandb/offline-run-20250814_133406-h6n7dll6
/home/dsriv/miniconda3/envs/llamagen_dummy/lib/python3.12/site-packages/torch/distributed/distributed_c10d.py:4807: UserWarning: No device id is provided via `init_process_group` or `barrier `. Using the current device set by the user. 
  warnings.warn(  # warn only once
[rank0]:[W814 13:34:06.479329532 ProcessGroupNCCL.cpp:5023] [PG ID 0 PG GUID 0 Rank 0]  using GPU 0 as device used by this process is currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect. You can specify device_id in init_process_group() to force use of a particular device.
[Rank 0][[34m2025-08-14 13:34:07[0m] Experiment directory: outputs/llamagen_b/20250814_13
[Rank 0][[34m2025-08-14 13:34:07[0m] Checkpoint directory: outputs/llamagen_b/20250814_13/checkpoints
[Rank 0][[34m2025-08-14 13:34:07[0m] Distributed environment: DistributedType.MULTI_GPU  Backend: nccl
Num processes: 2
Process index: 0
Local process index: 0
Device: cuda:0

Mixed precision type: bf16

[Rank 0][[34m2025-08-14 13:34:07[0m] GPU capacity: NVIDIA A100-SXM4-40GB (0) with 39.38GiB memory
[Rank 0][[34m2025-08-14 13:34:07[0m] GPU memory usage: NVIDIA A100-SXM4-40GB (0): 39.3812255859375 GiB capacity, 0.001953125 GiB peak, 0.004959533307915725% peak
[Rank 0][[34m2025-08-14 13:34:08[0m] Dataset contains 1,281,167 images (data/res256/imagenet_code_c2i_flip_ten_crop) with flip augmentation and 20 crop augmentation
[Rank 0][[34m2025-08-14 13:34:09[0m] GPT Parameters: 110,888,448
[Rank 0][[34m2025-08-14 13:34:10[0m] Image tokenizer is loaded
[Rank 0][[34m2025-08-14 13:34:10[0m] num decayed parameter tensors: 63, with 110,869,248 parameters
[Rank 0][[34m2025-08-14 13:34:10[0m] num non-decayed parameter tensors: 25, with 19,200 parameters
[Rank 0][[34m2025-08-14 13:34:10[0m] using fused AdamW: True
[Rank 0][[34m2025-08-14 13:34:10[0m] Training for 300 epochs starting at epoch 0...
[Rank 0][[34m2025-08-14 13:34:10[0m] Beginning epoch 0...
/home/dsriv/code/dynamic_tokenization/train_llamagen.py:274: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than torch.tensor(sourceTensor).
  average_grad_norm = torch.tensor(
/home/dsriv/code/dynamic_tokenization/train_llamagen.py:274: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than torch.tensor(sourceTensor).
  average_grad_norm = torch.tensor(
[Rank 0][[34m2025-08-14 13:34:46[0m] Epoch 00000000 | Step 00000100 | Loss 9.6816 | Time 0.3615s | Grad Norm 0.1343 | LR 0.00010 |  mem: 28% | pow: 355.542 W
[Rank 0][[34m2025-08-14 13:35:20[0m] Epoch 00000000 | Step 00000200 | Loss 9.5771 | Time 0.3370s | Grad Norm 0.1695 | LR 0.00010 |  mem: 28% | pow: 359.168 W
[Rank 0][[34m2025-08-14 13:35:54[0m] Epoch 00000000 | Step 00000300 | Loss 9.4520 | Time 0.3372s | Grad Norm 0.2114 | LR 0.00010 |  mem: 28% | pow: 354.603 W
[Rank 0][[34m2025-08-14 13:36:28[0m] Epoch 00000000 | Step 00000400 | Loss 9.3637 | Time 0.3375s | Grad Norm 0.2268 | LR 0.00010 |  mem: 28% | pow: 369.046 W
[Rank 0][[34m2025-08-14 13:37:01[0m] Epoch 00000000 | Step 00000500 | Loss 9.2818 | Time 0.3374s | Grad Norm 0.2662 | LR 0.00010 |  mem: 28% | pow: 364.925 W
[Rank 0][[34m2025-08-14 13:37:35[0m] Epoch 00000000 | Step 00000600 | Loss 9.2100 | Time 0.3373s | Grad Norm 0.3007 | LR 0.00010 |  mem: 28% | pow: 350.373 W
[Rank 0][[34m2025-08-14 13:38:09[0m] Epoch 00000000 | Step 00000700 | Loss 9.1341 | Time 0.3372s | Grad Norm 0.3241 | LR 0.00010 |  mem: 28% | pow: 345.264 W
[Rank 0][[34m2025-08-14 13:38:43[0m] Epoch 00000000 | Step 00000800 | Loss 9.0669 | Time 0.3372s | Grad Norm 0.3470 | LR 0.00010 |  mem: 28% | pow: 345.114 W
[Rank 0][[34m2025-08-14 13:39:16[0m] Epoch 00000000 | Step 00000900 | Loss 8.9989 | Time 0.3369s | Grad Norm 0.3754 | LR 0.00010 |  mem: 28% | pow: 339.514 W
[Rank 0][[34m2025-08-14 13:39:50[0m] Epoch 00000000 | Step 00001000 | Loss 8.9291 | Time 0.3368s | Grad Norm 0.3825 | LR 0.00010 |  mem: 28% | pow: 356.286 W
[Rank 0][[34m2025-08-14 13:40:24[0m] Epoch 00000000 | Step 00001100 | Loss 8.8681 | Time 0.3368s | Grad Norm 0.3999 | LR 0.00010 |  mem: 28% | pow: 352.063 W
[Rank 0][[34m2025-08-14 13:40:57[0m] Epoch 00000000 | Step 00001200 | Loss 8.8110 | Time 0.3367s | Grad Norm 0.3951 | LR 0.00010 |  mem: 28% | pow: 343.172 W
[Rank 0][[34m2025-08-14 13:41:31[0m] Epoch 00000000 | Step 00001300 | Loss 8.7624 | Time 0.3371s | Grad Norm 0.4059 | LR 0.00010 |  mem: 28% | pow: 364.193 W
[Rank 0][[34m2025-08-14 13:42:05[0m] Epoch 00000000 | Step 00001400 | Loss 8.7264 | Time 0.3370s | Grad Norm 0.3964 | LR 0.00010 |  mem: 28% | pow: 357.144 W
[Rank 0][[34m2025-08-14 13:42:38[0m] Epoch 00000000 | Step 00001500 | Loss 8.7005 | Time 0.3366s | Grad Norm 0.3975 | LR 0.00010 |  mem: 28% | pow: 343.934 W
[Rank 0][[34m2025-08-14 13:43:12[0m] Epoch 00000000 | Step 00001600 | Loss 8.6700 | Time 0.3365s | Grad Norm 0.3991 | LR 0.00010 |  mem: 28% | pow: 363.182 W
[Rank 0][[34m2025-08-14 13:43:46[0m] Epoch 00000000 | Step 00001700 | Loss 8.6380 | Time 0.3366s | Grad Norm 0.3910 | LR 0.00010 |  mem: 28% | pow: 346.587 W
[Rank 0][[34m2025-08-14 13:44:19[0m] Epoch 00000000 | Step 00001800 | Loss 8.6216 | Time 0.3370s | Grad Norm 0.3828 | LR 0.00010 |  mem: 28% | pow: 347.962 W
[Rank 0][[34m2025-08-14 13:44:53[0m] Epoch 00000000 | Step 00001900 | Loss 8.5952 | Time 0.3369s | Grad Norm 0.3881 | LR 0.00010 |  mem: 28% | pow: 364.973 W
[Rank 0][[34m2025-08-14 13:45:27[0m] Epoch 00000000 | Step 00002000 | Loss 8.5801 | Time 0.3370s | Grad Norm 0.3779 | LR 0.00010 |  mem: 28% | pow: 351.398 W
```

compile with aot-eager
```
The following values were not passed to `accelerate launch` and had defaults used instead:
	`--num_processes` was set to a value of `2`
	`--num_machines` was set to a value of `1`
	`--mixed_precision` was set to a value of `'no'`
	`--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
TorchDynamoPlugin(backend=<DynamoBackend.AOT_EAGER: 'AOT_EAGER'>, mode='default', fullgraph=False, dynamic=False, options=None, disable=False, use_regional_compilation=False)
TorchDynamoPlugin(backend=<DynamoBackend.AOT_EAGER: 'AOT_EAGER'>, mode='default', fullgraph=False, dynamic=False, options=None, disable=False, use_regional_compilation=False)
/home/dsriv/miniconda3/envs/llamagen_dummy/lib/python3.12/site-packages/torch/distributed/distributed_c10d.py:4807: UserWarning: No device id is provided via `init_process_group` or `barrier `. Using the current device set by the user. 
  warnings.warn(  # warn only once
[rank1]:[W814 14:09:17.176076808 ProcessGroupNCCL.cpp:5023] [PG ID 0 PG GUID 0 Rank 1]  using GPU 1 as device used by this process is currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect. You can specify device_id in init_process_group() to force use of a particular device.
wandb: Tracking run with wandb version 0.21.1
wandb: W&B syncing is set to `offline` in this directory. Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
wandb: Run data is saved locally in outputs/llamagen_b/20250814_14/wandb/offline-run-20250814_140917-8mtkp5sb
wandb:                                                                                
wandb: You can sync this run to the cloud by running:
wandb: wandb sync outputs/llamagen_b/20250814_14/wandb/offline-run-20250814_140917-8mtkp5sb
wandb: Find logs at: outputs/llamagen_b/20250814_14/wandb/offline-run-20250814_140917-8mtkp5sb/logs
wandb: Tracking run with wandb version 0.21.1
wandb: W&B syncing is set to `offline` in this directory. Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
wandb: Run data is saved locally in outputs/llamagen_b/20250814_14/wandb/offline-run-20250814_140918-ja620dvo
/home/dsriv/miniconda3/envs/llamagen_dummy/lib/python3.12/site-packages/torch/distributed/distributed_c10d.py:4807: UserWarning: No device id is provided via `init_process_group` or `barrier `. Using the current device set by the user. 
  warnings.warn(  # warn only once
[rank0]:[W814 14:09:18.309107634 ProcessGroupNCCL.cpp:5023] [PG ID 0 PG GUID 0 Rank 0]  using GPU 0 as device used by this process is currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect. You can specify device_id in init_process_group() to force use of a particular device.
[Rank 0][[34m2025-08-14 14:09:19[0m] Experiment directory: outputs/llamagen_b/20250814_14
[Rank 0][[34m2025-08-14 14:09:19[0m] Checkpoint directory: outputs/llamagen_b/20250814_14/checkpoints
[Rank 0][[34m2025-08-14 14:09:19[0m] Distributed environment: DistributedType.MULTI_GPU  Backend: nccl
Num processes: 2
Process index: 0
Local process index: 0
Device: cuda:0

Mixed precision type: bf16

[Rank 0][[34m2025-08-14 14:09:19[0m] GPU capacity: NVIDIA A100-SXM4-40GB (0) with 39.38GiB memory
[Rank 0][[34m2025-08-14 14:09:19[0m] GPU memory usage: NVIDIA A100-SXM4-40GB (0): 39.3812255859375 GiB capacity, 0.001953125 GiB peak, 0.004959533307915725% peak
[Rank 0][[34m2025-08-14 14:09:20[0m] Dataset contains 1,281,167 images (data/res256/imagenet_code_c2i_flip_ten_crop) with flip augmentation and 20 crop augmentation
[Rank 0][[34m2025-08-14 14:09:21[0m] GPT Parameters: 110,888,448
[Rank 0][[34m2025-08-14 14:09:22[0m] Image tokenizer is loaded
[Rank 0][[34m2025-08-14 14:09:22[0m] num decayed parameter tensors: 63, with 110,869,248 parameters
[Rank 0][[34m2025-08-14 14:09:22[0m] num non-decayed parameter tensors: 25, with 19,200 parameters
[Rank 0][[34m2025-08-14 14:09:22[0m] using fused AdamW: True
[Rank 0][[34m2025-08-14 14:09:23[0m] Training for 300 epochs starting at epoch 0...
[Rank 0][[34m2025-08-14 14:09:23[0m] Beginning epoch 0...
[rank0]:W0814 14:09:24.473000 275697 site-packages/torch/_logging/_internal.py:1154] [0/0] Profiler function <class 'torch.autograd.profiler.record_function'> will be ignored
[rank1]:W0814 14:09:24.814000 275698 site-packages/torch/_logging/_internal.py:1154] [0/0] Profiler function <class 'torch.autograd.profiler.record_function'> will be ignored
/home/dsriv/code/dynamic_tokenization/train_llamagen.py:273: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than torch.tensor(sourceTensor).
  average_grad_norm = torch.tensor(
/home/dsriv/code/dynamic_tokenization/train_llamagen.py:273: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than torch.tensor(sourceTensor).
  average_grad_norm = torch.tensor(
[Rank 0][[34m2025-08-14 14:10:19[0m] Epoch 00000000 | Step 00000100 | Loss 9.6815 | Time 0.5643s | Grad Norm 0.1344 | LR 0.00010 |  mem: 25% | pow: 342.224 W
[Rank 0][[34m2025-08-14 14:10:55[0m] Epoch 00000000 | Step 00000200 | Loss 9.5770 | Time 0.3596s | Grad Norm 0.1696 | LR 0.00010 |  mem: 25% | pow: 339.999 W
[Rank 0][[34m2025-08-14 14:11:31[0m] Epoch 00000000 | Step 00000300 | Loss 9.4518 | Time 0.3599s | Grad Norm 0.2116 | LR 0.00010 |  mem: 25% | pow: 342.478 W
[Rank 0][[34m2025-08-14 14:12:07[0m] Epoch 00000000 | Step 00000400 | Loss 9.3636 | Time 0.3596s | Grad Norm 0.2274 | LR 0.00010 |  mem: 25% | pow: 370.703 W
[Rank 0][[34m2025-08-14 14:12:43[0m] Epoch 00000000 | Step 00000500 | Loss 9.2816 | Time 0.3598s | Grad Norm 0.2647 | LR 0.00010 |  mem: 25% | pow: 352.712 W
[Rank 0][[34m2025-08-14 14:13:19[0m] Epoch 00000000 | Step 00000600 | Loss 9.2098 | Time 0.3598s | Grad Norm 0.3006 | LR 0.00010 |  mem: 25% | pow: 351.398 W
[Rank 0][[34m2025-08-14 14:13:55[0m] Epoch 00000000 | Step 00000700 | Loss 9.1338 | Time 0.3599s | Grad Norm 0.3258 | LR 0.00010 |  mem: 25% | pow: 349.805 W
[Rank 0][[34m2025-08-14 14:14:31[0m] Epoch 00000000 | Step 00000800 | Loss 9.0662 | Time 0.3600s | Grad Norm 0.3435 | LR 0.00010 |  mem: 25% | pow: 344.316 W
[Rank 0][[34m2025-08-14 14:15:07[0m] Epoch 00000000 | Step 00000900 | Loss 8.9982 | Time 0.3598s | Grad Norm 0.3764 | LR 0.00010 |  mem: 25% | pow: 347.903 W
[Rank 0][[34m2025-08-14 14:15:43[0m] Epoch 00000000 | Step 00001000 | Loss 8.9282 | Time 0.3598s | Grad Norm 0.3846 | LR 0.00010 |  mem: 25% | pow: 354.641 W
[Rank 0][[34m2025-08-14 14:16:19[0m] Epoch 00000000 | Step 00001100 | Loss 8.8675 | Time 0.3601s | Grad Norm 0.4023 | LR 0.00010 |  mem: 25% | pow: 354.52 W
[Rank 0][[34m2025-08-14 14:16:55[0m] Epoch 00000000 | Step 00001200 | Loss 8.8104 | Time 0.3601s | Grad Norm 0.3948 | LR 0.00010 |  mem: 25% | pow: 356.286 W
[Rank 0][[34m2025-08-14 14:17:31[0m] Epoch 00000000 | Step 00001300 | Loss 8.7619 | Time 0.3601s | Grad Norm 0.4065 | LR 0.00010 |  mem: 25% | pow: 346.289 W
[Rank 0][[34m2025-08-14 14:18:07[0m] Epoch 00000000 | Step 00001400 | Loss 8.7260 | Time 0.3600s | Grad Norm 0.3963 | LR 0.00010 |  mem: 25% | pow: 339.891 W
[Rank 0][[34m2025-08-14 14:18:43[0m] Epoch 00000000 | Step 00001500 | Loss 8.7001 | Time 0.3598s | Grad Norm 0.3987 | LR 0.00010 |  mem: 25% | pow: 337.123 W
[Rank 0][[34m2025-08-14 14:19:19[0m] Epoch 00000000 | Step 00001600 | Loss 8.6695 | Time 0.3599s | Grad Norm 0.3987 | LR 0.00010 |  mem: 25% | pow: 351.117 W
[Rank 0][[34m2025-08-14 14:19:55[0m] Epoch 00000000 | Step 00001700 | Loss 8.6376 | Time 0.3598s | Grad Norm 0.3909 | LR 0.00010 |  mem: 25% | pow: 345.647 W
[Rank 0][[34m2025-08-14 14:20:31[0m] Epoch 00000000 | Step 00001800 | Loss 8.6212 | Time 0.3599s | Grad Norm 0.3836 | LR 0.00010 |  mem: 25% | pow: 356.612 W
[Rank 0][[34m2025-08-14 14:21:07[0m] Epoch 00000000 | Step 00001900 | Loss 8.5948 | Time 0.3599s | Grad Norm 0.3872 | LR 0.00010 |  mem: 25% | pow: 337.657 W
[Rank 0][[34m2025-08-14 14:21:43[0m] Epoch 00000000 | Step 00002000 | Loss 8.5796 | Time 0.3599s | Grad Norm 0.3767 | LR 0.00010 |  mem: 25% | pow: 353.041 W
```

compile with inductor
```
The following values were not passed to `accelerate launch` and had defaults used instead:
	`--num_processes` was set to a value of `2`
	`--num_machines` was set to a value of `1`
	`--mixed_precision` was set to a value of `'no'`
	`--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
TorchDynamoPlugin(backend=<DynamoBackend.INDUCTOR: 'INDUCTOR'>, mode='default', fullgraph=False, dynamic=False, options=None, disable=False, use_regional_compilation=False)
TorchDynamoPlugin(backend=<DynamoBackend.INDUCTOR: 'INDUCTOR'>, mode='default', fullgraph=False, dynamic=False, options=None, disable=False, use_regional_compilation=False)
/home/dsriv/miniconda3/envs/llamagen_dummy/lib/python3.12/site-packages/torch/distributed/distributed_c10d.py:4807: UserWarning: No device id is provided via `init_process_group` or `barrier `. Using the current device set by the user. 
  warnings.warn(  # warn only once
[rank1]:[W814 14:01:54.737532968 ProcessGroupNCCL.cpp:5023] [PG ID 0 PG GUID 0 Rank 1]  using GPU 1 as device used by this process is currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect. You can specify device_id in init_process_group() to force use of a particular device.
wandb: Tracking run with wandb version 0.21.1
wandb: W&B syncing is set to `offline` in this directory. Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
wandb: Run data is saved locally in outputs/llamagen_b/20250814_14/wandb/offline-run-20250814_140155-ywx8i6qx
wandb:                                                                                
wandb: You can sync this run to the cloud by running:
wandb: wandb sync outputs/llamagen_b/20250814_14/wandb/offline-run-20250814_140155-ywx8i6qx
wandb: Find logs at: outputs/llamagen_b/20250814_14/wandb/offline-run-20250814_140155-ywx8i6qx/logs
wandb: Tracking run with wandb version 0.21.1
wandb: W&B syncing is set to `offline` in this directory. Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
wandb: Run data is saved locally in outputs/llamagen_b/20250814_14/wandb/offline-run-20250814_140155-emd5mcd3
/home/dsriv/miniconda3/envs/llamagen_dummy/lib/python3.12/site-packages/torch/distributed/distributed_c10d.py:4807: UserWarning: No device id is provided via `init_process_group` or `barrier `. Using the current device set by the user. 
  warnings.warn(  # warn only once
[rank0]:[W814 14:01:56.845767606 ProcessGroupNCCL.cpp:5023] [PG ID 0 PG GUID 0 Rank 0]  using GPU 0 as device used by this process is currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect. You can specify device_id in init_process_group() to force use of a particular device.
[Rank 0][[34m2025-08-14 14:01:56[0m] Experiment directory: outputs/llamagen_b/20250814_14
[Rank 0][[34m2025-08-14 14:01:56[0m] Checkpoint directory: outputs/llamagen_b/20250814_14/checkpoints
[Rank 0][[34m2025-08-14 14:01:56[0m] Distributed environment: DistributedType.MULTI_GPU  Backend: nccl
Num processes: 2
Process index: 0
Local process index: 0
Device: cuda:0

Mixed precision type: bf16

[Rank 0][[34m2025-08-14 14:01:56[0m] GPU capacity: NVIDIA A100-SXM4-40GB (0) with 39.38GiB memory
[Rank 0][[34m2025-08-14 14:01:56[0m] GPU memory usage: NVIDIA A100-SXM4-40GB (0): 39.3812255859375 GiB capacity, 0.001953125 GiB peak, 0.004959533307915725% peak
[Rank 0][[34m2025-08-14 14:01:57[0m] Dataset contains 1,281,167 images (data/res256/imagenet_code_c2i_flip_ten_crop) with flip augmentation and 20 crop augmentation
[Rank 0][[34m2025-08-14 14:01:58[0m] GPT Parameters: 110,888,448
[Rank 0][[34m2025-08-14 14:01:59[0m] Image tokenizer is loaded
[Rank 0][[34m2025-08-14 14:01:59[0m] num decayed parameter tensors: 63, with 110,869,248 parameters
[Rank 0][[34m2025-08-14 14:01:59[0m] num non-decayed parameter tensors: 25, with 19,200 parameters
[Rank 0][[34m2025-08-14 14:01:59[0m] using fused AdamW: True
[Rank 0][[34m2025-08-14 14:02:03[0m] Training for 300 epochs starting at epoch 0...
[Rank 0][[34m2025-08-14 14:02:03[0m] Beginning epoch 0...
[rank1]:W0814 14:02:04.392000 269404 site-packages/torch/_logging/_internal.py:1154] [0/0] Profiler function <class 'torch.autograd.profiler.record_function'> will be ignored
[rank0]:W0814 14:02:04.450000 269403 site-packages/torch/_logging/_internal.py:1154] [0/0] Profiler function <class 'torch.autograd.profiler.record_function'> will be ignored
/home/dsriv/code/dynamic_tokenization/train_llamagen.py:273: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than torch.tensor(sourceTensor).
  average_grad_norm = torch.tensor(
/home/dsriv/code/dynamic_tokenization/train_llamagen.py:273: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than torch.tensor(sourceTensor).
  average_grad_norm = torch.tensor(
[Rank 0][[34m2025-08-14 14:02:43[0m] Epoch 00000000 | Step 00000100 | Loss 9.6810 | Time 0.4047s | Grad Norm 0.1354 | LR 0.00010 |  mem: 19% | pow: 383.596 W
[Rank 0][[34m2025-08-14 14:02:59[0m] Epoch 00000000 | Step 00000200 | Loss 9.5782 | Time 0.1588s | Grad Norm 0.1700 | LR 0.00010 |  mem: 19% | pow: 393.958 W
[Rank 0][[34m2025-08-14 14:03:15[0m] Epoch 00000000 | Step 00000300 | Loss 9.4584 | Time 0.1593s | Grad Norm 0.2041 | LR 0.00010 |  mem: 19% | pow: 407.012 W
[Rank 0][[34m2025-08-14 14:03:31[0m] Epoch 00000000 | Step 00000400 | Loss 9.3806 | Time 0.1593s | Grad Norm 0.2103 | LR 0.00010 |  mem: 19% | pow: 387.514 W
[Rank 0][[34m2025-08-14 14:03:47[0m] Epoch 00000000 | Step 00000500 | Loss 9.3164 | Time 0.1594s | Grad Norm 0.2178 | LR 0.00010 |  mem: 19% | pow: 339.756 W
[Rank 0][[34m2025-08-14 14:04:03[0m] Epoch 00000000 | Step 00000600 | Loss 9.2676 | Time 0.1595s | Grad Norm 0.2351 | LR 0.00010 |  mem: 19% | pow: 413.101 W
[Rank 0][[34m2025-08-14 14:04:19[0m] Epoch 00000000 | Step 00000700 | Loss 9.2165 | Time 0.1596s | Grad Norm 0.2427 | LR 0.00010 |  mem: 19% | pow: 393.849 W
[Rank 0][[34m2025-08-14 14:04:35[0m] Epoch 00000000 | Step 00000800 | Loss 9.1720 | Time 0.1595s | Grad Norm 0.2528 | LR 0.00010 |  mem: 19% | pow: 382.804 W
[Rank 0][[34m2025-08-14 14:04:51[0m] Epoch 00000000 | Step 00000900 | Loss 9.1294 | Time 0.1595s | Grad Norm 0.2644 | LR 0.00010 |  mem: 19% | pow: 389.968 W
[Rank 0][[34m2025-08-14 14:05:07[0m] Epoch 00000000 | Step 00001000 | Loss 9.0890 | Time 0.1595s | Grad Norm 0.2573 | LR 0.00010 |  mem: 19% | pow: 386.113 W
[Rank 0][[34m2025-08-14 14:05:23[0m] Epoch 00000000 | Step 00001100 | Loss 9.0539 | Time 0.1595s | Grad Norm 0.2711 | LR 0.00010 |  mem: 19% | pow: 388.945 W
[Rank 0][[34m2025-08-14 14:05:39[0m] Epoch 00000000 | Step 00001200 | Loss 9.0133 | Time 0.1597s | Grad Norm 0.2657 | LR 0.00010 |  mem: 19% | pow: 412.446 W
[Rank 0][[34m2025-08-14 14:05:55[0m] Epoch 00000000 | Step 00001300 | Loss 8.9722 | Time 0.1598s | Grad Norm 0.2697 | LR 0.00010 |  mem: 19% | pow: 362.591 W
[Rank 0][[34m2025-08-14 14:06:11[0m] Epoch 00000000 | Step 00001400 | Loss 8.9359 | Time 0.1596s | Grad Norm 0.2811 | LR 0.00010 |  mem: 19% | pow: 418.244 W
[Rank 0][[34m2025-08-14 14:06:26[0m] Epoch 00000000 | Step 00001500 | Loss 8.9059 | Time 0.1593s | Grad Norm 0.2859 | LR 0.00010 |  mem: 19% | pow: 395.712 W
[Rank 0][[34m2025-08-14 14:06:42[0m] Epoch 00000000 | Step 00001600 | Loss 8.8708 | Time 0.1595s | Grad Norm 0.2862 | LR 0.00010 |  mem: 19% | pow: 385.135 W
[Rank 0][[34m2025-08-14 14:06:58[0m] Epoch 00000000 | Step 00001700 | Loss 8.8358 | Time 0.1598s | Grad Norm 0.2903 | LR 0.00010 |  mem: 19% | pow: 378.918 W
[Rank 0][[34m2025-08-14 14:07:14[0m] Epoch 00000000 | Step 00001800 | Loss 8.8158 | Time 0.1599s | Grad Norm 0.2899 | LR 0.00010 |  mem: 19% | pow: 390.617 W
[Rank 0][[34m2025-08-14 14:07:30[0m] Epoch 00000000 | Step 00001900 | Loss 8.7875 | Time 0.1598s | Grad Norm 0.2990 | LR 0.00010 |  mem: 19% | pow: 370.032 W
[Rank 0][[34m2025-08-14 14:07:46[0m] Epoch 00000000 | Step 00002000 | Loss 8.7681 | Time 0.1598s | Grad Norm 0.2861 | LR 0.00010 |  mem: 19% | pow: 417.464 W
```

Environment:
```
absl-py==2.3.1
accelerate==1.10.0
annotated-types==0.7.0
antlr4-python3-runtime==4.9.3
certifi==2025.8.3
charset-normalizer==3.4.3
click==8.2.1
contourpy==1.3.3
cycler==0.12.1
filelock==3.13.1
fonttools==4.59.1
fsspec==2024.6.1
gitdb==4.0.12
GitPython==3.1.45
grpcio==1.74.0
hf-xet==1.1.7
huggingface-hub==0.34.4
idna==3.10
Jinja2==3.1.4
kiwisolver==1.4.9
Markdown==3.8.2
MarkupSafe==2.1.5
matplotlib==3.10.5
mpmath==1.3.0
networkx==3.3
numpy==2.1.2
nvidia-cublas-cu12==12.6.4.1
nvidia-cuda-cupti-cu12==12.6.80
nvidia-cuda-nvrtc-cu12==12.6.77
nvidia-cuda-runtime-cu12==12.6.77
nvidia-cudnn-cu12==9.10.2.21
nvidia-cufft-cu12==11.3.0.4
nvidia-cufile-cu12==1.11.1.6
nvidia-curand-cu12==10.3.7.77
nvidia-cusolver-cu12==11.7.1.2
nvidia-cusparse-cu12==12.5.4.2
nvidia-cusparselt-cu12==0.7.1
nvidia-ml-py==12.575.51
nvidia-nccl-cu12==2.27.3
nvidia-nvjitlink-cu12==12.6.85
nvidia-nvtx-cu12==12.6.77
omegaconf==2.3.0
opencv-python==4.12.0.88
packaging==25.0
pandas==2.3.1
pillow==11.0.0
platformdirs==4.3.8
protobuf==6.31.1
psutil==7.0.0
pydantic==2.11.7
pydantic_core==2.33.2
pynvml==12.0.0
pyparsing==3.2.3
python-dateutil==2.9.0.post0
pytorch-fid==0.3.0
pytz==2025.2
PyYAML==6.0.2
requests==2.32.4
safetensors==0.6.2
scipy==1.16.1
seaborn==0.13.2
sentry-sdk==2.35.0
setuptools==78.1.1
six==1.17.0
smmap==5.0.2
sympy==1.13.3
tensorboard==2.20.0
tensorboard-data-server==0.7.2
torch==2.8.0+cu126
torchvision==0.23.0+cu126
tqdm==4.67.1
triton==3.4.0
typing-inspection==0.4.1
typing_extensions==4.12.2
tzdata==2025.2
urllib3==2.5.0
wandb==0.21.1
Werkzeug==3.1.3
wheel==0.45.1
```

**Comment by dsrivastavv** [comment]
@bdhirsh Were you able to repro this issue?
